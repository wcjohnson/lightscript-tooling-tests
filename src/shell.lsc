exec = require('child_process').exec

export run(cmd) ->
  new Promise! (resolve, reject) ->
    exec! cmd, (err, stdout, stderr) ->
      if err:
        reject({err, stdout, stderr})
      else:
        resolve({stdout, stderr})

export snapshot(expect, cmd) ->
  run! cmd
    .then! ({stdout, stderr}) ->
      expect({stdout, stderr, succeeded: true, err: null}).toMatchSnapshot()
    .catch! ({err, stdout, stderr}) ->
      expect({stdout, stderr, succeeded: false, err}).toMatchSnapshot()

export successfulSnapshot(expect, cmd) ->
  run! cmd
    .then! ({stdout, stderr}) ->
      expect({stdout, stderr, succeeded: true, err: null}).toMatchSnapshot()
