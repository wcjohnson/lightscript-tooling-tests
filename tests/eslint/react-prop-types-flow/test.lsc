import { snapshot } from '../../../lib/shell'
path = require('path')

let testName = __dirname.split(path.sep)
now testName = testName[testName.length - 1]

test! testName, ->
  expect~snapshot! `$(npm bin)/eslint -c ${__dirname}/linter-config.json ${__dirname}/lintThis.lsc`
