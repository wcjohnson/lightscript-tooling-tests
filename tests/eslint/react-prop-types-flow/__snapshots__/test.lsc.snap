// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-prop-types-flow 1`] = `
Object {
  "exitStatus": 1,
  "stderr": "",
  "stdout": "
/Users/wcj/dev/lightscript/lightscript-tooling-tests/tests/eslint/react-prop-types-flow/lintThis.lsc
  13:29  error  'data' is missing in props validation  react/prop-types

âœ– 1 problem (1 error, 0 warnings)

",
  "succeeded": false,
}
`;
